From 04b4744396316562c0fdcff18e7ece24f5078cef Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Tue, 15 May 2018 17:35:21 +0800
Subject: [PATCH] ARM: dts: rk3288: enable ddr frequency scaling for linux evb

Enable dmc and dfi nodes to support ddr frequency
scaling on rk3288-evb-linux board.

Change-Id: I578b7dcace09a337abf9c7f640786db588ff9406
Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
---
 arch/arm/boot/dts/rk3288-evb-act8846.dts     |  9 +++++++
 arch/arm/boot/dts/rk3288-evb-rk808-linux.dts | 25 ++++++++++++++++++++
 2 files changed, 34 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288-evb-act8846.dts b/arch/arm/boot/dts/rk3288-evb-act8846.dts
index 2ad38486525c..280f3c385d13 100644
--- a/arch/arm/boot/dts/rk3288-evb-act8846.dts
+++ b/arch/arm/boot/dts/rk3288-evb-act8846.dts
@@ -189,6 +189,15 @@
 	cpu0-supply = <&vdd_cpu>;
 };
 
+&dfi {
+	status = "okay";
+};
+
+&dmc {
+	center-supply = <&vdd_log>;
+	status = "okay";
+};
+
 &edp {
 	vcc-supply = <&vdd10_lcd>;
 	vccio-supply = <&vcc18_lcd>;
diff --git a/arch/arm/boot/dts/rk3288-evb-rk808-linux.dts b/arch/arm/boot/dts/rk3288-evb-rk808-linux.dts
index f32b6ba90d05..397d46b7e65b 100644
--- a/arch/arm/boot/dts/rk3288-evb-rk808-linux.dts
+++ b/arch/arm/boot/dts/rk3288-evb-rk808-linux.dts
@@ -128,12 +128,33 @@
 	};
 
 	/delete-node/ sdmmc-regulator;
+
+	vdd_log: vdd-logic {
+		compatible = "pwm-regulator";
+		rockchip,pwm_id = <1>;
+		rockchip,pwm_voltage = <1100000>;
+		pwms = <&pwm1 0 25000 1>;
+		regulator-name = "vcc_log";
+		regulator-min-microvolt = <860000>;
+		regulator-max-microvolt = <1360000>;
+		regulator-always-on;
+		regulator-boot-on;
+	};
 };
 
 &cpu0 {
 	cpu0-supply = <&vdd_cpu>;
 };
 
+&dfi {
+	status = "okay";
+};
+
+&dmc {
+	center-supply = <&vdd_log>;
+	status = "okay";
+};
+
 &edp {
 	status = "okay";
 };
@@ -467,6 +488,10 @@
 	status = "okay";
 };
 
+&pwm1 {
+	status = "okay";
+};
+
 &pinctrl {
 	backlight {
 		bl_en: bl-en {
-- 
2.35.3

